README.md

```diff
@@ -2,40 +2,120 @@
-* Pre-loading and rendering of non-visible content asynchronously to prepare it for future display (Example: improving latency to show content not currently displayed but predicted to be soon, but *without jank*. Think search as you type, tabbed UIs, hero element clicks.)
-* Layout measurement (examples: responsive design or animation setup)
+* Avoid loading [\[2\]](#foot-notes) and rendering work for content not visible
+  to the user.
+* Support [user-agent features](https://github.com/WICG/display-locking/blob/master/user-agent-features.md)
+  and all layout algorithms (e.g. responsive design, flexbox, grid) for this
+  content
 
-## Motivation & background
+* Support developer-controlled pre-loading, pre-rendering and measurement of
+  content without having to fully render it to the screen
 
-Faster web page loads and interactions directly improve the user experience of the web. On the other hand, web sites each year are larger and more complex than the last, in part because they support more and more use cases, and contain more information. This leads to pages with a lot of DOM, and since the DOM presently renders atomically, it inherently takes more and more time to render on the same machine.
-
-For these reasons, web developers need ways to reduce loading and rendering time of web apps that have a lot of DOM. Two common techniques are to mark non-visible DOM as "invisible" [\[3\]](#foot-notes), or to use virtualization [\[4\]](#foot-notes). Browser implementors also want to reduce loading and rendering time of web apps. Common techniques to do so include adding caching of rendering state [\[5\]](#foot-notes), and avoiding rendering work [\[6\]](#foot-notes) for content that is not visible.
+The following use-cases motivate this work:
 
-These techniques can work in many cases but have some drawbacks and limitations. These include:
-a. [\[3\]](#foot-notes) and [\[4\]](#foot-notes) usually means that such content is not available to user-agent features. Also, content that is merely placed offscreen may or may not have rendering cost (it depends on browser heuristics), which makes the technique unreliable.
-b. Caching intermediate rendering state is [hard work](https://martinfowler.com/bliki/TwoHardThings.html), and often has performance limitations and cliffs that are not obvious to developers. Similarly, relying on the browser to avoid rendering for content that is clipped out or not visible is sometimes not reliable, as it's hard for the browser to efficiently detect what content is actually visible.
+* Fast display of large HTML documents (examples: HTML one-page spec, other long
+  documents)
+* Deep links and searchability into pages with hidden content (example: mobile
+  Wikipedia)
+* Scrollers with a large amount of content, without resorting to virtualization
+  (examples: twitter feed, codemirror documents)
+* Pre-loading and rendering of non-visible content asynchronously to prepare it
```

Comment:
How about re-phrasing this bullet as:

* Single-page app transitions

Reply:
Done.